rasa_nlu_examples/fallback/fasttext_language_fallback.py

```diff
@@ -0,0 +1,309 @@
+    ]
+
+    def __init__(self, component_config: Optional[Dict[Text, Any]] = None) -> None:
+        super().__init__(component_config)
+        if "cache_dir" not in component_config.keys():
+            raise ValueError(
+                "You need to set `cache_dir` for the FasttextLanguageFallbackClassifier."
+            )
+        if "file" not in component_config.keys():
+            raise ValueError(
+                "You need to set `file` for the FasttextLanguageFallbackClassifier."
+            )
+        if "intent_triggered" not in component_config.keys():
+            raise ValueError(
+                "You need to set `intent_triggered` for the FasttextLanguageFallbackClassifier."
+            )
+        if "expected_language" not in component_config.keys():
+            raise ValueError(
+                "You need to set `language` for the FasttextLanguageFallbackClassifier."
+            )
+        if component_config["expected_language"] not in self.language_list:
+            raise ValueError(
+                "You have specified a unsupported language. \n"
+                "See https://fasttext.cc/docs/en/language-identification.html for supported languages."
+            )
+        path = os.path.join(component_config["cache_dir"], component_config["file"])
+        if not os.path.exists(component_config["cache_dir"]):
+            raise FileNotFoundError(
+                f"It seems that the cache dir {component_config['cache_dir']} does not exists. Please check config."
+            )
+        if not os.path.exists(path):
+            raise FileNotFoundError(
+                f"It seems that file {path} does not exists. Please check config."
+            )
+
+        self.model = fasttext.load_model(path=path)
+        self.expected_language = component_config["expected_language"]
+        self.min_chars = component_config["min_chars"]
+        self.min_tokens = component_config["min_tokens"]
+        self.threshold = component_config["threshold"]
+        self.intent_triggered = component_config["intent_triggered"]
+        self.protected_intents = (
+            component_config["protected_intents"]
+            if "protected_intents" in component_config
+            else self.defaults["protected_intents"]
+        )
+
+    @classmethod
+    def required_components(cls) -> List[Type[Component]]:
+        return [IntentClassifier, Tokenizer]
+
+    def process(self, message: Message, **kwargs: Any) -> None:
+        """Process an incoming message."""
+        if message.data[INTENT]["name"] in self.protected_intents:
+            logger.debug("The model detected a protected intent, won't overwrite.")
+            return
+
+        if self._not_enough_text(message):
+            logger.debug(
+                "There's not enough text to do proper language identification."
+            )
+            return
+
+        proba_dict = {
+            k: v for k, v in zip(*self.model.predict(message.get(TEXT), k=20))
```

Comment:
I guess `k` refers to the language code and `v` to the confidence value? If so, I would rename the variables accordingly.